### ISSUE TITLE
Switch in Dialog renders large ripple

### ISSUE THREAD

**Issue by MatthiasF999** [material-components/material-components-web-react#719]
When I add a Switch to a Dialog, the ::before element is huge.
![dialogwithswitch](https://user-images.githubusercontent.com/11466334/53903784-e1b1f200-4044-11e9-9fcb-d7ddf4a0d023.png)

I believe this is because the switch is initialized inside a closed dialog, but this behavior does not occur when using the material-components-web (example here: https://codepen.io/matthiasf999/pen/qvRomz).

**Comment by mgr34** [comment]
I had a similiar issue #657

**Comment by moog16** [comment]
@mgr34 yup this sounds like an issue from MDC Web about a year ago. I think we added layout then. Fix is similar to #657: to add a layout method in the ripple/index.tsx on an update.

**Comment by MatthiasF999** [comment]
The MDC Web version does not contain a layout method. That was the first thing I looked for. The work around does not help either.

**Comment by moog16** [comment]
The layout method needs to be called on the ripple method.
